lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -573,15 +540,7 @@ fromTip genesisHash tip = case getPoint (getTipPoint tip) of
         { slotNo = Point.blockPointSlot blk
         , blockHeight = fromBlockNo $ getLegacyTipBlockNo tip
         , headerHash = fromCardanoHash $ Point.blockPointHash blk
-        -- TODO
-        -- We only use the parentHeaderHash in the
-        -- 'Cardano.Wallet.Network.BlockHeaders' chain follower only required for
-        -- Jörmungandr, this is therefore useless to have in 'normal' BlockHeader
-        --
-        -- Yet, since we also serialize these to the database, this requires
-        -- some non-trivial changes. Not fixing this right now is also a
-        -- possibility.
-        , parentHeaderHash = W.Hash "parentHeaderHash - unused in Shelley"
+        , parentHeaderHash = W.Hash ""
```

Comment:
Removing `parentHeaderHash` would be nice, but is this `"parentHeaderHash - unused in Shelley"` to `""` change safe? It couldn't affect compatibility with existing DBs somehow?

Reply:
No, I think we only do logic with the BlockHeader after converting it to a `Point CardanoBlock`, so this should be fine.

Reply:
I tried removing `parentHeaderHash` but then found that it is sort of used in `Cardano.Wallet.restoreBlocks` !

Reply:
Ah!

Reply:
Perhaps you could try removing parentHeaderHash? To keep DB compatibility, just leave the field there, but don't use it.

Reply:
I suspect it might make sense to do with/after #2750 if we can replace `BlockHeader` with `Point CardanoBlock` and be able to persist the Origin point in a nicer way. Currently I _think_ rollbacks to `Origin` might break:

```
G 0 1 <-- Wallet has checkpoints at genesis, slot 0 and slot 1
Node asks wallet to rollback to G
Wallet believes genesis has slotNo 0:
G 0
Node asks wallet to re-apply block in slot 0:
G 0 0 <- Wrong!
```

Although...

1. This invariant is nice to have, particularly as we lack great chain-following testing
https://github.com/input-output-hk/cardano-wallet/blob/8d1b692fec0c77e488a724dba97c9fa51ab9f4e5/lib/core/src/Cardano/Wallet.hs#L888-L893
2. I am a bit concerned about how to handle backward compatibility regarding how `Origin` is encoded https://github.com/input-output-hk/cardano-wallet/blob/914668d4e40db6a0b999869c4c4f5222fa6884d0/lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs#L392-L398